python-lib/text_extraction/__init__.py

```diff
@@ -54,4 +54,137 @@ def extract_text_content(file_bytes, extension, with_pandoc):
             return file_bytes.decode()
         
         return text
-            
\ No newline at end of file
+
+
+def extract_text_chunks(filename, file_bytes, extension, with_pandoc):
+    if extension == "pdf":
+        pdf_pages = pdfium.PdfDocument(file_bytes)
+        return [
+            {
+                'file': filename,
+                'text': page.get_textpage().get_text_range(),
+                'id': page_id + 1,
+                'metadata': {"page": page_id + 1},
+                'error_message': ""
+            }
+            for page_id, page in enumerate(pdf_pages)
+        ]
+    elif extension == "doc":
+        raise ValueError("'doc' files are not supported, try to convert them to docx.")
+    elif extension == "md":
+        return extract_markdown_chunks(file_bytes.decode(), filename)
```

Comment:
Is this necessary with the pandoc conversion below ? In other words, is the md to md pandoc conversion no-op ?